lib/index.js

```diff
@@ -171,6 +170,25 @@ function getPkgIds(filteredPkgs) {
   return filteredPkgs.map(({ latest, name }) => `${name}@${latest}`).join(' ');
 }
 
+/**
+ * Filters out dependencies with locked versions.
+ *
+ * @param {Object.<string, string>} deps - List of dependencies.
+ * @returns {Object} List of dependencies excluding locked semver versions.
+ */
+function removeLockedDependencies(deps) {
+  let depsToUpgrade = {};
+
+  for (const name of Object.keys(deps)) {
+    const version = deps[name];
+    const firstChar = version.charAt(0);
+    if (firstChar === '^' || firstChar === '~') {
+      depsToUpgrade = { ...depsToUpgrade, [name]: version };
```

Comment:
Good observation, but... there is a concept of hidden classes. Objects don't like when their properties are added or removed. By destructuring the original object with new properties we create a new one, thus avoiding deoptimization.
Node.js optimizes objects under the hood, and if we continuously change object structure, the optimization would give up and slow down execution. Since we could have many dependencies to process, this could become a bottleneck.
<img width="1413" alt="Screen Shot 2021-04-05 at 10 24 44 AM" src="https://user-images.githubusercontent.com/7528575/141166634-bdef9dab-d8c5-4a80-9bf3-de82e9900331.png">